Client/Frontend/Share/CreatePDFActivity.swift

```diff
@@ -32,20 +30,8 @@ class CreatePDFActivity: UIActivity {
     }
     
     override func perform() {
-        webView.createPDF { [weak self] result in
-            dispatchPrecondition(condition: .onQueue(.main))
-            guard let self = self else {
-                return
-            }
-            switch result {
-            case .success(let data):
-                self.callback(data)
-                self.activityDidFinish(true)
-            case .failure(let error):
-                log.error("Failed to create PDF with error: \(error)")
-                self.activityDidFinish(false)
-            }
-        }
+        callback()
+        activityDidFinish(true)
```

Comment:
We should be passing back the success/failure to call back into the `UIActivity`

Reply:
I followed here what we do with other activities, like `AddFeedToBraveNewsActivity`

Reply:
`AddFeedToBraveNewsActivity` is different, it's not created unless success is the only option, in this case the activity is created but it can fail based on the executed task. You can simply do `callback: () -> Bool` instead and do `activityDidFinish(callback())` then return false when it fails in createPDF result closure